sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java

```diff
@@ -44,15 +47,15 @@ public void register(final @NotNull IHub hub, final @NotNull SentryOptions optio
       try {
         Class.forName("androidx.lifecycle.DefaultLifecycleObserver");
         Class.forName("androidx.lifecycle.ProcessLifecycleOwner");
-        watcher =
-            new LifecycleWatcher(
-                hub,
-                this.options.getSessionTrackingIntervalMillis(),
-                this.options.isEnableSessionTracking(),
-                this.options.isEnableAppLifecycleBreadcrumbs());
-        ProcessLifecycleOwner.get().getLifecycle().addObserver(watcher);
-
-        options.getLogger().log(SentryLevel.DEBUG, "AppLifecycleIntegration installed.");
+        if (MainThreadChecker.isMainThread()) {
+          addObserver(hub);
+          options.getLogger().log(SentryLevel.DEBUG, "AppLifecycleIntegration installed.");
```

Comment:
`l`: don't we also want to have this log message in the `else` below?

Reply:
good catch, the `else` came later for just getting a review and showing an example and forgot to move things, thx